### PR TITLE
Update memory.md

### LINKED ISSUE #13778: Add a note regarding setting size limit on shared caches
The docs references a "shared" memory cache in DI early on and this leads to a pitfall when users try to set a size limit on the shared instance later on. When a size limit is imposed on a cache, all entries must specify a size when being added. This can lead to issues since the users may not have full control on what uses the shared cache. For example, EFCore will use the shared cache and does not specify a size, and the application will throw InvalidOperationException if a size limit is imposed on the shared cache. We should add a note early in the documentation so that it's clear we recommend against setting a size limit on a shared memory cache and instead tell users to create a private instance for caches with a size limit. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 13f5f8f9-18dd-9c12-9ed1-fcc6b44ba7fd
* Version Independent ID: 640b4f36-5886-7b69-1c95-cfc8574a02fd
* Content: [Cache in-memory in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-2.2#using-imemorycache)
* Content Source: [aspnetcore/performance/caching/memory.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/performance/caching/memory.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-performance**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #13778

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-2.2&branch=pr-en-us-13786#using-imemorycache)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@mmitche please review.

File: aspnetcore/performance/caching/memory.md [hunk 1]
```diff
@@ -51,7 +51,10 @@ Use `System.Runtime.Caching`/`MemoryCache` as a compatibility bridge when portin
 
 ## Using IMemoryCache
 
-In-memory caching is a *service* that's referenced from your app using [Dependency Injection](../../fundamentals/dependency-injection.md). Call `AddMemoryCache` in `ConfigureServices`:
+> [!WARNING]
+> Using a *shared* memory cache from [Dependency Injection](xref:fundamentals/dependency-injection) and calling `SetSize`, `Size`, or `SizeLimit` to limit cache size can cause the app to fail. When a size limit is set on a cache, all entries must specify a size when being added. This can lead to issues since developers may not have full control on what uses the shared cache. For example, Entity Framework Core uses the shared cache and does not specify a size. If an app sets a cache size limit and uses EF Core, the app throw's an `InvalidOperationException`.
```
Comment (mmitche) [inline aspnetcore/performance/caching/memory.md]: "the app throw's an" -> "the app will throw an"

Comment (mmitche) [reply aspnetcore/performance/caching/memory.md]: I'd also add a bit about a proper solution in this case (construct your own cache singleton), and link to the sample below (`MyMemoryCache`) which implements it.

Comment (Rick-Anderson) [reply aspnetcore/performance/caching/memory.md]: > "the app throw's an" -> "the app will throw an"

Per MS style guide, avoid future tense - it doesn't MT (machine translate) well.

> add a bit about a proper solution

Added, please review

**Comment by Rick-Anderson** [comment]
@mmitche how's this?

File: aspnetcore/performance/caching/memory.md (hunk 1 above)
Comment (mmitche) [reply aspnetcore/performance/caching/memory.md]: It should be "throws" rather than "throw's" then, I think. It's not possessive.

**Pull request merged**